### DISCUSSION TITLE
Is it possible to scan only specific commit?

### DISCUSSION THREAD

**Discussion by inatale** [zricethezav/gitleaks#783] [General]
I am trying to use gitleaks in pre-receive hook in GitLab but gitleaks return scan results for all commit tree but I want to scan only fresh commits. I tried to use --log-opts="${commithash}" but without any luck - result was the same: full scan of a repository. Could you please advise somehing?

PS. git log -p ${commithash} returns only info specific to the commit

PPS. And for example --log-opts="-n 5" works correctly.

**Comment by mota-lhd** [comment]
hi ! I use this : --log-opts="--max-count=1 $COMMIT_SHA1"

you need to pay attention on how you clone your repo (shallow or full). if shallow, you need to fetch :
git fetch --depth=$DEPTH --force origin "$BRANCH:remotes/origin/$BRANCH"

**Comment by zricethezav** [comment]
> hi ! I use this : --log-opts="--max-count=1 $COMMIT_SHA1"

Is a good option, `--log-opts="-1 {commit}"` also works.
